code/modules/mob/living/silicon/robot/robot_modules.dm

```diff
@@ -484,6 +520,8 @@
 	moduleselect_icon = "service"
 	special_light_key = "service"
 	hat_offset = 0
+	whitelisted_reagents.Add(list("orangejuice" , "tomatojuice")) //Add here anything it can synthetize regardless of emag status

```

Comment:
So I either hardcode the list to each borg or I add it somewhere later? The intention was to have each borg to be able to have their own whitelist, on top of a global whitelist.

Reply:
add it to `Initialize(mapload)`

so
```DM
/obj/item/robot_module/Initialize(mapload)
    . = ..()
    whitelisted_reagents += list
```

If you already generate a list though I'm not sure why you don't just include it in the original list. The inefficiency is generated from the creation of the list, not so much when you add things to the list.

I'm not entirely sure what the goal here is though so I don' t want to suggest you in the wrong direction. I'd consider tying that aspect to the feature on reagents instead of the module.

Reply:
What I wanted to add was like.. for example.. Every borg could learn how to synth orangejuice by default (A global whitelist), however, ServiceBorg can learn tomatojuice, but not the other borgs. So it would be ONLY whitelisted to ServiceBorg, so MedicalBorg could not do tomatojuice. 
But if MedicalBorg got emagged, it could then make the tomatojuice, because it is not considering the whitelist anymore.